CourseManager/Views/CoursesView.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  ContentView.swift
+//  CourseManager
+//
+//  Created by Summer Crow on 2022-09-29.
+//
+
+import SwiftUI
+
+struct CoursesView: View {
+    @State private var showingSheet = false
+    @State private var selectedCourseID = ""
+
+    @ObservedObject private var viewModel = CoursesViewModel()
+
+    var body: some View {
+        NavigationView {
+            listOfCourses
+                .onAppear {
+                    self.viewModel.fetchData()
+                }
+
+                .toolbar {
+                    Button {
+                        showingSheet.toggle()
+                    } label: {
+                        Image(systemName: "plus")
+                            .font(.title2)
+                    }.sheet(isPresented: $showingSheet) {
+                        AddOrEditCourseSheet(course: CourseInformation(id: "", name: "", url: "", gitHubURL: "", status: .todo, dateOfCompletion: Date(), comments: ""))
+                    }
+                }
+                .navigationTitle("Courses")
+        }
+    }
+
+    var listOfCourses: some View {
+        List {
+            ForEach(viewModel.courses) { course in
+
+                NavigationLink {
+                    if let courseID = course.id {
+                        CourseDetailsView(courseID: courseID)
+                    }
+                } label: {
+                    HStack {
+                        Text(course.name)
+                        Spacer()
+                        Group {
+                            course.status == .complete ? Status.complete.image : course.status == .inProgress ? Status.inProgress.image : Status.todo.image
```

Comment:
`course.status.image` or if there are two different statuses, then add init to transfer from one model to another

Reply:
Wow I really overcomplicated this one. I've replaced the code with the line you suggested and it works just the same.